### ISSUE TITLE
Clarify <iframe> behavior in <svg:use>

### ISSUE THREAD

**Issue by emilio** [w3c/svgwg#876]
https://www.w3.org/TR/SVG2/struct.html#UseShadowTree:

> Within a [use-element shadow tree](https://www.w3.org/TR/SVG2/struct.html#TermUseElementShadowTree), ‘[script](https://www.w3.org/TR/SVG2/interact.html#ScriptElement)’ elements are inert (do not execute); ‘[audio](https://www.w3.org/TR/SVG2/embedded.html#HTMLElements)’ and ‘[video](https://www.w3.org/TR/SVG2/embedded.html#HTMLElements)’ elements are subject to the limitations specified in the [Multimedia](https://www.w3.org/TR/SVG2/struct.html#UseMultimedia) section.
>
> > Previous versions of SVG restricted the contents of the shadow tree to SVG graphics elements. This specification allows any valid SVG document subtree to be cloned. Cloning non-graphical content, however, will not usually have any visible effect.

Ok, so `<script>` is inert. But `<iframe>` isn't, which means I can load an `<iframe>` and execute script. That seems fairly weird. Is it intended? If not, what should the behavior of `<iframe>` inside `<svg:use>` be?

**Comment by BigBadaboom** [comment]
`<iframe>` is not an SVG element.

**Comment by emilio** [comment]
Right, I mean when `<iframe>` is inside `<foreignObject>`

**Comment by karlcow** [comment]
see discussions on https://bugzilla.mozilla.org/show_bug.cgi?id=1754522

**Comment by Psychpsyo** [comment]
Note: This also extends to all `on...` attributes, like `onload` or `onerror` on an `<img>`.
These scripts currently run in Firefox. (although I assume they're not supposed to, even if the current spec's wording doesn't account for it)

**Comment by mozfreddyb** [comment]
Can you file a separate issue for this? If this is just one browser, you'd be best advised to file it against the browser directly (eg https://bugzilla.mozilla.org/home)

**Comment by Psychpsyo** [comment]
> Can you file a separate issue for this? If this is just one browser, you'd be best advised to file it against the browser directly (eg https://bugzilla.mozilla.org/home)

Well, it's only one browser because only one browser implements that part of the spec currently. And that part of the spec unambiguously says that the behavior in that browser is correct, so it's more of a potential oversight in the spec. (and it's the exact same oversight that this bug is about.)

**Comment by karlcow** [comment]
There is a resolution on #875

**Comment by karlcow** [comment]
The current prose of the specification is:

https://w3c.github.io/svgwg/svg2-draft/struct.html#:~:text=The-,shadow%20tree%20is%20open,-(inspectable%20by%20script

> The shadow tree is open (inspectable by script), but read-only. Any attempt to directly modify the elements, attributes, and other nodes in the shadow tree must throw a NoModificationAllowedError.
>
> **Within a [use-element shadow tree](https://w3c.github.io/svgwg/svg2-draft/struct.html#TermUseElementShadowTree), ‘[script](https://w3c.github.io/svgwg/svg2-draft/interact.html#ScriptElement)’ elements are inert (do not execute).**
> 
> Previous versions of SVG restricted the contents of the shadow tree to SVG graphics elements. This specification allows any valid SVG document subtree to be cloned. Cloning non-graphical content, however, will not usually have any visible effect.
> 

It has changed since @emilio opening the issue.

**Comment by emilio** [comment]
> The shadow tree is open (inspectable by script), but read-only. Any attempt to directly modify the elements, attributes, and other nodes in the shadow tree must throw a NoModificationAllowedError.

I don't think this matches any browser. Strong preference to spec that the shadow tree is closed. Otherwise this needs sweeping changes around all DOM APIs which has performance implications and is a huge whack-a-mole.

**Comment by karlcow** [comment]
We need better prose and testing.

**Comment by Psychpsyo** [comment]
> We need better prose and testing.

Can I write those?

**Comment by Psychpsyo** [comment]
Wrote some tests: https://psychpsyo.com/web/testing/svg/useElement.html
Results:
The shadow root is closed in all browsers. (WebKit, Chromium, Firefox)
Iframes exist in Firefox and all scripts, in or outside of them, do execute.
(the tests are written to verify that scripts don't run, but that's just me picking one option to see what Firefox currently does)
